spec/lib/alpha_taxonomy/taxon_linker_spec.rb

```diff
@@ -0,0 +1,42 @@
+require "rails_helper"
+
+RSpec.describe AlphaTaxonomy::TaxonLinker do
+  describe "#run!" do
+    def stub_taxon_fetch(title:, content_id:)
+      allow(Services.content_store).to receive(:content_item!).with(
+        AlphaTaxonomy::TaxonPresenter.new(title: title).base_path
+      ).and_return("content_id" => content_id)
+    end
+
+    def stub_content_item_fetch(base_path:, content_id:)
+      allow(Services.content_store).to receive(:content_item!).with(
+        base_path
+      ).and_return("content_id" => content_id)
+    end
+
+    it "creates taxon links based on the grouped mappings" do
+      import_file = AlphaTaxonomy::ImportFile.new
+      allow(import_file).to receive(:grouped_mappings).and_return(
+        "/a-foo-content-item" => ["Foo Taxon"],
+        "/a-foo-bar-content-item" => ["Foo Taxon", "Bar Taxon"]
+      )
+      allow(AlphaTaxonomy::ImportFile).to receive(:new).and_return(import_file)
```

Comment:
The stubbing makes this a little hard to follow. Perhaps it is clearer if you make `TaxonLinker.new.run!` take a `AlphaTaxonomy::ImportFile` as an argument. Then you could do this:

```ruby
import_file = AlphaTaxonomy::ImportFile.new
import_file.stub(grouped_mappings: [...])
AlphaTaxonomy::TaxonLinker.new.run!(import_file)
```